### PR TITLE
Bugfix/incremental rendering is same node text

### LINKED ISSUE #423: TypeError: n.batchProps is not a function
## Are you reporting a bug or a feature?

 - [x] Bug

## Expected Behavior

React Todo example at https://axa-ch.github.io/patterns-library/react.html should work.

## Actual Behavior

Does not even start.

Maybe it's a cache or production build problem.

## Steps to Reproduce the Problem

1. go to https://axa-ch.github.io/patterns-library/react.html
 
## Failure Logs

```sh
all-demos.js:1 TypeError: n.batchProps is not a function
    at t.value (all-demos.js:1)
    at t.value (all-demos.js:1)
    at commitLifeCycles (all-demos.js:1)
    at t (all-demos.js:1)
    at w (all-demos.js:1)
    at x (all-demos.js:1)
    at h (all-demos.js:1)
    at f (all-demos.js:1)
    at Object.updateContainer (all-demos.js:1)
    at all-demos.js:1
u @ all-demos.js:1
t @ all-demos.js:1
w @ all-demos.js:1
x @ all-demos.js:1
h @ all-demos.js:1
f @ all-demos.js:1
updateContainer @ all-demos.js:1
(anonymous) @ all-demos.js:1
unbatchedUpdates @ all-demos.js:1
qi @ all-demos.js:1
render @ all-demos.js:1
bu @ all-demos.js:1
(anonymous) @ all-demos.js:1
```

### LINKED ISSUE #424: Broken Button With React after master merge
`<AXAButtonReact onAxaClick={this.handleClick} key={1} >{this.state.isToggleOn ? 'ON' : 'OFF'}</AXAButtonReact>` does not update (always ON is shown. Is ONLY appeared after master merge as it was tested before mergin and worked correctly

on localhost

### PR DESCRIPTION
**Pull request by AndyOGo**
Fixes #424 .
Fixes #423 

Changes proposed in this pull request:

 - incremental rendering and morphing approach
 - fixes accordion and header

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] This change requires a documentation update
 
 # How Has This Been Tested?
 
 
 
 # Checklist:
 
 - [x] My code follows the style guidelines of this project
 - [x] I have performed a self-review of my own code
 - [x] I have commented my code, particularly in hard-to-understand areas
 - [x] I have made corresponding changes to the documentation
 - [x] My changes generate no new warnings
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/js/abstract/base-component.js [hunk 1]
```diff
@@ -350,7 +351,11 @@ export default class BaseComponent extends HTMLElement {
           this._lightDOMRefs.forEach((ref) => {
             // Important: Once the light DOM is live it shouldn't be moved out
             // instead make sure to clone it for incremental updates
-            const refClone = ref.cloneNode(false);
+            const refClone = ref.cloneNode(true);
```
Comment (AndyOGo) [inline src/js/abstract/base-component.js]: @LucaMele 
Just this line seems to fix our issue, but by cloning including the whole subtree we will get some performance hits (to be honest I don't see them being observable).

File: src/js/abstract/is-same-node-once.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+/**
+ * Make sure that another piece of code is/can managing that part of the DOM tree.
+ *
+ * @link https://github.com/choojs/nanomorph#caching-dom-elements
+ * @param node
+ */
+export function isSameNodeOnce(node) {
+  return;
+  // @todo: react text-node updates break if we do this!
+  if (node.nodeType === TEXT_NODE) {
```
Comment (AndyOGo) [inline src/js/abstract/is-same-node-once.js]: @LucaMele 
This check did **NOT** fixed the broken accordion and header.
So not an option and the whole function seems to stay pointless.
Cause I really don't know yet what is causing our issue and why my fix works I'm uncertain if should delete it now...

Comment (AndyOGo) [reply src/js/abstract/is-same-node-once.js]: `isSameNodeOnce` did work for Custom-elements only.
But broke for integration with react.
So may we could say that `deepClone` and `isSameNode` optimisation are configurable?

File: src/js/abstract/base-component.js (hunk 1 above)
Comment (LucaMele) [reply src/js/abstract/base-component.js]: true.. and that wont work in IE

File: src/js/with-react.jsx [hunk 3]
```diff
@@ -40,9 +39,9 @@ const isEventFilter = (key) => {
  * );
  */
 const withReact = (WebComponent, { pure = true, passive = false } = {}) => {
-  const name = functionName(WebComponent);
-  const displayName = `${name}React`;
-  const WCTagName = dasherize(name);
+  const { tagName } = (new WebComponent());
```
Comment (LucaMele) [inline src/js/with-react.jsx]: wow ok i see

**Comment by LucaMele** [comment]
IE11 myaxa button and language dropdown is not rendered inside the `AXA-HEADER-META-RIGHT`

bildschirmfoto 2018-05-04 um 14 44 42

http://localhost:8080/organisms.html#o-header

File: src/js/with-react.jsx (hunk 3 above)
Comment (AndyOGo) [reply src/js/with-react.jsx]: but breaks with **Uncaught TypeError: Illegal constructor** 😞 
screen shot 2018-05-04 at 15 01 47

Comment (LucaMele) [reply src/js/with-react.jsx]: :(

Comment (AndyOGo) [reply src/js/with-react.jsx]: Okay, I quickly test two cases:
1. Instantiating `AXAButton` before it's definition by ` window.customElements.define('axa-button', AXAButton);` -> **Uncaught TypeError: Illegal constructor**
2.  Instantiating `AXAButton` after it's definition -> works

File: src/js/with-react.jsx [hunk 4]
```diff
@@ -108,10 +107,23 @@ const withReact = (WebComponent, { pure = true, passive = false } = {}) => {
...
-      const { props: { children } } = this;
+      const { props: { children }, handleRef } = this;
+
+      // IMPORTANT:
+      // the Custom Element can only be instantiated as soon as it is registered in CustomElementRegistry
+      // which in turn is deferred after DOMReady
+      // hence it's tagName can only be resolved lazily
+      // ref: https://developer.mozilla.org/en-US/docs/Web/API/CustomElementRegistry
+      if (!WCTagName) {
+        const node = new WebComponent();
```
Comment (AndyOGo) [inline src/js/with-react.jsx]: Instanciating Custom Elements after the are defined is suuuuper important.
If we try to do it before they will always throw **Invalid Constructor** errors...

**Pull request merged**